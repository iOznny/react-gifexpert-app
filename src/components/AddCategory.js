import React, { useState } from 'react'
import PropTypes from 'prop-types';

export const AddCategory = ({ setCategories }) => {

    // Inicio del input
    const [inputValue, setInputValue] = useState('');

    // Cambiar valor del input
    const handleInputChange = (e) => {        
        setInputValue(e.target.value);
        console.log('handleInputChange llamado');
    };

    // Prevenir posteo de form y agregar nuevo valor al arreglo
    const handleSubmit = (e) => {                
        e.preventDefault();

        if (inputValue.trim().length > 2) {
            setCategories(cat => [inputValue, ...cat]);
            setInputValue('');
        }
    }

    return (  
        <form onSubmit={ handleSubmit }>
            <p>{ inputValue }</p>
            <input 
                type="text"
                value={ inputValue } 
                onChange={ handleInputChange }
            />
        </form>
    )
}

// Requerir el prop-types
AddCategory.propTypes = {
    setCategories: PropTypes.func.isRequired
}
