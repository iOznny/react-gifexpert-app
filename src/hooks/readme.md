- Para normar un HOOK se utiliza al inicio de cada archivo la palabra (use) ejemplo useFetchGifs.js indicando que el siguiente archivo es un Hook.
- Los hooks no son mas que simples funciones.


    * Custom Hook: Permiten extraer la logica de algun componente para la reutilizaci√≥n permitiendo extraerla de tal manera permita ser sencillo utilizarla nuevamente *