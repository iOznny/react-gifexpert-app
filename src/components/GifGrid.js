import React from 'react';
import PropTypes from 'prop-types';

// Hooks
import { GifGridItem } from './GifGridItem';
import { useFetchGifs } from '../hooks/useFetchGifs'

export const GifGrid = ({ category }) => {

    const { data: images, loading } = useFetchGifs(category);
    
    return (
        <>    
            <h5>{ category }</h5>    
            { loading && <p className="animate__animated animate__flash">Cargando...</p> }

            <div className="card-grid">
                {
                    images.map(img => (<GifGridItem key={ img.id } { ...img } />))
                }
            </div>
        </>
    )
}

GifGrid.propTypes = {
    category: PropTypes.string.isRequired
}