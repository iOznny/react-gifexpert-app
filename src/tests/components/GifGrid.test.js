import React from 'react';
import { shallow } from "enzyme"
import '@testing-library/jest-dom';

import { GifGrid } from "../../components/GifGrid"
import { useFetchGifs } from '../../hooks/useFetchGifs';
jest.mock('../../hooks/useFetchGifs');

describe('GifGrid', () => {
    
    const category = 'One';

    test('Show successful', () => {
        useFetchGifs.mockReturnValue({
            data: [],
            loading: true
        });

        const wrapper = shallow(<GifGrid category={ category } />);
        expect(wrapper).toMatchSnapshot();
    });

    // Cuando el useFetch regresa la informacion.
    test('Show items when load images useFetchGifs', () => {
        const gifs = [
            {
                id: 'abc',
                title: 'Calquier cosa',
                url: 'https://localhost.com/awdawdawd'
            },
            {
                id: 'abcd',
                title: 'Calquier cosa',
                url: 'https://localhost.com/awdawdawd'
            }
        ];

        useFetchGifs.mockReturnValue({
            data: gifs,
            loading: false
        });

        const wrapper = shallow(<GifGrid category={ category } />);
        //expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('p').exists()).toBe(false);
        expect(wrapper.find('GifGridItem').length).toBe(gifs.length);
    });
    
});
