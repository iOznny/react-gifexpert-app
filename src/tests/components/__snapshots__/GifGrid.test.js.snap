// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GifGrid Show successful 1`] = `
<Fragment>
  <h5>
    One
  </h5>
  <p
    className="animate__animated animate__flash"
  >
    Cargando...
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;
